external/wakaama/core/data.c

```diff
@@ -439,15 +440,27 @@ int lwm2m_data_parse(lwm2m_uri_t * uriP,
         if (*dataP == NULL) return 0;
         (*dataP)->id = uriP->resourceId;
         (*dataP)->type = LWM2M_TYPE_STRING;
-        return prv_setBuffer(*dataP, buffer, bufferLen);
+        res = prv_setBuffer(*dataP, buffer, bufferLen);
```

Comment:
Could you please explain the difference with respect to the previous version? I think the memory is freed in the older version of the code, when the prv_setBuffer returns 0. The concern here is with respect to code maintainence, especially because data.c is wakaama core, which we merge from eclipse.